File: server/com/FightforSeven/handler/LoginHandler.java
package com.FightforSeven.handler;

import java.util.List;

import com.FightforSeven.manager.AccountManager;
import com.FightforSeven.model.SocketModel;
import com.FightforSeven.protocol.TypeProtocol;
import com.FightforSeven.server.MainServer;
import com.FightforSeven.protocol.AreaProtocol;
import com.FightforSeven.protocol.CommandProtocol;

import io.netty.channel.ChannelHandlerContext;

/**
 * @author 作者：littleredhat
 * @date 创建时间：2016年11月21日 上午12:20:43
 * @version 1.0
 *
 */
public class LoginHandler extends BaseHandler{
	private volatile int num = 0;
	
	@Override
	public int getType() {
		return TypeProtocol.TYPE_LOGIN;
	}

	@Override
	public void dispatch(ChannelHandlerContext ctx, SocketModel message) {
		switch (message.getArea()) {
		case AreaProtocol.Area_LoginRequest:
			loginResponse(ctx, message);
			break;
		default:
			break;
		}
	}

	public int checkAccount(SocketModel request) {
		List<String> message = request.getMessage();
		String account = message.get(0);
		String password = message.get(1);
		System.out.println(++num + " from client");
		return AccountManager.getInstance().checkAccount(account, password);
	}

	public void loginResponse(ChannelHandlerContext ctx, SocketModel request) {
		SocketModel response = new SocketModel();
		int command = checkAccount(request);
		
		response.setType(TypeProtocol.TYPE_LOGIN);
		response.setArea(AreaProtocol.Area_LoginResponse);
		response.setCommand(command);
		response.setMessage(request.getMessage());
		ctx.writeAndFlush(response);
		
		if (command == CommandProtocol.Login_Succeed) {
			MainServer.getInstance().channel2account.put(ctx.channel(), request.getMessage().get(0));  //记录登录成功的连接对应的账号
			MainServer.getInstance().account2channel.put(request.getMessage().get(0), ctx.channel());  //记录登录成功的账号对应的连接
		}
	}
	
}

File: client/Server/Handler/LoginHandler.cs
using UnityEngine;
using UnityEngine.SceneManagement;
using System;
using System.Collections;
using System.Collections.Generic;

public class LoginHandler : BaseHandler
{
    public override int type
    {
        get
        {
            return TypeProtocol.TYPE_LOGIN;
        }
    }

    public override void Dispatch(int command, List<String> message)
    {
        switch (command)
        {
            case CommandProtocol.Login_InvalidMessage:
                MessageManager.instance.ShowLog("请重新登录...");
                break;
            case CommandProtocol.Login_InvalidAccount:
                MessageManager.instance.ShowLog("账号不存在...");
                break;
            case CommandProtocol.Login_InvalidPassword:
                MessageManager.instance.ShowLog("密码错误...");
                break;
            case CommandProtocol.Login_Failed:
                MessageManager.instance.ShowLog("登录失败...");
                break;
            case CommandProtocol.Login_Succeed:
                //登录成功，如果存在角色则加载，不存在则创建
                SocketModel getRoleRequset = new SocketModel();
                getRoleRequset.SetType(TypeProtocol.TYPE_GET_ROLE);
                getRoleRequset.SetArea(AreaProtocol.Area_GetRoleRequest);
                getRoleRequset.SetCommand(0);
                List<string> msg = new List<string>();
                msg.Add("");
                getRoleRequset.SetMessage(msg);
                MainClient.Instance.SendMsg(getRoleRequset);
                break;
        }
    }
}
